air/src/virtual_column.rs

```diff
@@ -113,10 +113,166 @@ impl<F: Field> VirtualPairCol<F> {
         Expr: PrimeCharacteristicRing + Mul<F, Output = Expr>,
         Var: Into<Expr> + Copy,
     {
-        let mut result = self.constant.into();
-        for (column, weight) in &self.column_weights {
-            result += column.get(preprocessed, main).into() * *weight;
-        }
-        result
+        self.column_weights
+            .iter()
+            .fold(self.constant.into(), |acc, (col, w)| {
+                acc + col.get(preprocessed, main).into() * *w
```

Comment:
Nit: Slightly clearer:
```suggestion
            .fold(self.constant.into(), |acc, &(col, w)| {
                acc + col.get(preprocessed, main).into() * w
```